classes/patreon_routing.php

```diff
@@ -112,13 +120,63 @@ function parse_request( &$wp ) {
 				
 			}
 			
-			if( array_key_exists( 'patreon-unlock-post', $wp->query_vars ) ) {
+			if( array_key_exists( 'patreon-direct-unlock', $wp->query_vars ) ) {
+				
+				$final_redirect = wp_login_url();
 				
-				// First slap the noindex header so search engines wont index this page:
-				header( 'X-Robots-Tag: noindex, nofollow' );
-				 
-				// Make sure browsers dont cache this
-				header( 'cache-control: no-cache, must-revalidate, max-age=0' );
+				if( isset( $wp->query_vars['patreon-direct-unlock'] ) ) {
+					
+					if( !( isset($GLOBALS['patreon_enable_direct_unlocks']) AND $GLOBALS['patreon_enable_direct_unlocks'] )
+					) {
+						
+						// No locking level set for this content or the site. No point in locking. Redirect to post.
+						$final_redirect = add_query_arg( 'patreon_message', 'patreon_direct_unlocks_not_turned_on', $final_redirect );
+						wp_redirect( $final_redirect );
+						
+						exit;	
+					}					
+					
+					$patreon_level = $wp->query_vars['patreon-direct-unlock'];
+					$redirect = base64_decode( urldecode( $wp->query_vars['patreon-redirect'] ) );
+					
+					if( !$patreon_level OR $patreon_level == '' OR $patreon_level == 0 ) {
+						$patreon_level = 1;
+					}
+					
+					$client_id = get_option( 'patreon-client-id', false );
+				
+					if( !$client_id ) {
+						
+						// No client id, no point in being here. Make it go with an error.
+						
+						$final_redirect = add_query_arg( 'patreon_message', 'patreon_cant_login_api_error_credentials', $final_redirect );
+						
+						wp_redirect( $final_redirect );
+						exit;
+						
+					}
+					
+					$post = false;
+					
+					// If post id set, get the post 
+					if( isset( $wp->query_vars['patreon-post-id'] ) ) {
+						$post = get_post( $wp->query_vars['patreon-post-id'] );
```

Comment:
spacing around this block

Reply:
We had had decided to leave one liners without spacing as i remember?

Reply:
Ah, yes you're correct